packages/core/src/defer/instructions.ts

```diff
@@ -110,18 +121,46 @@ export const DEFER_BLOCK_CONFIG = new InjectionToken<DeferBlockConfig>(
   ngDevMode ? 'DEFER_BLOCK_CONFIG' : '',
 );
 
+function shouldTriggerWhenOnServer(injector: Injector) {
+  const isServer = !isPlatformBrowser(injector);
+  const isIncrementalHydrationEnabled = injector.get(IS_INCREMENTAL_HYDRATION_ENABLED, false, {
+    optional: true,
+  });
+  return isServer && isIncrementalHydrationEnabled;
```

Comment:
We could add an early return here to avoid retrieving value from an injector unnecessarily: 
```suggestion
function shouldTriggerWhenOnServer(injector: Injector) {
  if (isPlatformBrowser(injector)) {
    return false; // do not trigger when in a browser
  }
  return injector.get(IS_INCREMENTAL_HYDRATION_ENABLED, false, {
    optional: true,
  });
```